server/src/main/java/org/elasticsearch/search/lookup/SourceLookup.java

```diff
@@ -126,6 +126,13 @@ private static Tuple<XContentType, Map<String, Object>> sourceAsMapAndType(Bytes
         return XContentHelper.convertToMap(source, false);
     }
 
+    /**
+     * Will there <strong>never</strong> be a {@code _source}?
+     */
+    public boolean alwaysEmpty() {
+        return sourceProvider.alwaysEmpty();
+    }
+
```

Comment:
Can we do this via the FieldDataContext somehow? Putting it on the SourceLookup feels a bit weird to me.

Reply:
I can move it, but, like, it's the source lookup that is empty. I think maybe moving and renaming it is a good idea. I just don't know the right spot.